src/liballoc/collections/btree/set.rs

```diff
@@ -1055,6 +1086,46 @@ impl<'a, T> IntoIterator for &'a BTreeSet<T> {
     }
 }
 
+/// An iterator produced by calling `drain_filter` on BTreeSet.
+#[unstable(feature = "btree_drain_filter", issue = "42849")]
+pub struct DrainFilter<'a, T>(
+    btree_map::DrainFilter<'a, T, (), Box<dyn 'a + FnMut(&T, &mut ()) -> bool>>,
+)
+where
+    T: 'a + Ord;
+
+#[unstable(feature = "btree_drain_filter", issue = "42849")]
+impl<'a, T> Drop for DrainFilter<'a, T>
+where
+    T: 'a + Ord,
+{
+    fn drop(&mut self) {
+        self.for_each(drop);
+    }
+}
+
+#[unstable(feature = "btree_drain_filter", issue = "42849")]
+impl<'a, T> fmt::Debug for DrainFilter<'a, T>
+where
+    T: 'a + Ord + fmt::Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_tuple("DrainFilter").field(&0).finish()
```

Comment:
Won't this always print literally "DrainFilter(0)"? Perhaps you meant to forward it -- `self.0.fmt(f)`